### PR TITLE
Add type definitions handling

### PR DESCRIPTION
**Pull request by jakubtiteo**

### CODE REVIEW

**Review by Rongix — COMMENTED** [review]
Nice, working as expected if typedefs are in the same file as config class, otherwise it generates empty class / ommits field that uses typedef from import.
Can we fix it in this PR?

**Review by Rongix — APPROVED** [review]

File: packages/theme_tailor/lib/src/generator/theme_tailor_generator.dart [hunk 1]
```diff
@@ -254,8 +273,37 @@ class _TailorClassASTVisitor extends SimpleAstVisitor {
...
 }
+
+List<CompilationUnit> _getLibrariesCompilationUnits(
+    List<LibraryElement> libraries) {
+  return libraries
+      .map(_getParsedLibraryResultFromElement)
+      .whereNotNull()
+      .map((lib) => lib.units.map((u) => u.unit))
+      .flattened
+      .toList();
```
Comment (Rongix) [inline packages/theme_tailor/lib/src/generator/theme_tailor_generator.dart:287]:
```suggestion
      .expand((lib) => lib.units.map((u) => u.unit))
      .toList();
```

File: packages/theme_tailor/lib/src/generator/theme_tailor_generator.dart [hunk 2]
```diff
@@ -254,8 +273,37 @@ class _TailorClassASTVisitor extends SimpleAstVisitor {
...
+}
+
+ParsedLibraryResult? _getParsedLibraryResultFromElement(Element element) {
+  final library = element.library!;
+  final parsedLibrary = library.session.getParsedLibraryByElement(library);
+  if (parsedLibrary is ParsedLibraryResult) {
+    return parsedLibrary;
+  } else {
+    return null;
+  }
```
Comment (Rongix) [inline packages/theme_tailor/lib/src/generator/theme_tailor_generator.dart:297]:
```suggestion
  final library = element.library;
  final parsedLibrary = library?.session.getParsedLibraryByElement(library);
  if (parsedLibrary is ParsedLibraryResult) {
    return parsedLibrary;
  } else {
    return null;
  }
```

**Review by Rongix — COMMENTED** [review]
Tiny suggestions below, 
Also I think it would be beneficial to move on of the typedefs to separate file and then import it.
e.g. 'typedef_example_lib.dart' with ListOfColor

**Review by Axot017 — APPROVED** [review]

**Pull request merged**